### PR TITLE
Customizable timeouts

### PR DESCRIPTION
**Pull request by PhilippeChepy**
Add the ability to define custom API clients timeouts.

For Exoscale builder, the default was raised from 5mn to 10mn.
For post-processor, the default is the previous hard-coded value (1h).

### Tests

Tests:

```
➜ make test    
/usr/local/go/bin/go test                      \
        -race                       \
        -mod vendor                 \
        -timeout 15s \
        -parallel 3 -count=1 -failfast       \
        github.com/exoscale/packer-plugin-exoscale/builder/exoscale github.com/exoscale/packer-plugin-exoscale/post-processor/exoscale-import
ok      github.com/exoscale/packer-plugin-exoscale/builder/exoscale     0.139s
ok      github.com/exoscale/packer-plugin-exoscale/post-processor/exoscale-import       0.097s
```

Acceptance tests:

```
➜ make test-acc
PACKER_ACC=1 /usr/local/go/bin/go test         \
        -race                       \
        -timeout 60m                \
        --tags=testacc              \
        -v -parallel 3 -count=1 -failfast       \
        github.com/exoscale/packer-plugin-exoscale/builder/exoscale github.com/exoscale/packer-plugin-exoscale/post-processor/exoscale-import
=== RUN   TestAccBuilder
2022/01/31 16:50:51 ui: Build ID: c7s0amq967th0l6cv530
2022/01/31 16:50:51 ui: Creating SSH key
2022/01/31 16:50:52 ui: Creating Compute instance
2022/01/31 16:51:11 ui: Using SSH communicator to connect: 185.19.28.65
2022/01/31 16:51:11 [INFO] Waiting for SSH, up to timeout: 5m0s
2022/01/31 16:51:11 ui: Waiting for SSH to become available...
2022/01/31 16:51:26 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 185.19.28.65:22: i/o timeout
2022/01/31 16:51:31 [INFO] Attempting SSH connection to 185.19.28.65:22...
2022/01/31 16:51:31 [DEBUG] reconnecting to TCP connection for SSH
2022/01/31 16:51:31 [DEBUG] handshaking with SSH
2022/01/31 16:51:32 [DEBUG] handshake complete!
2022/01/31 16:51:32 [DEBUG] Opening new ssh session
2022/01/31 16:51:33 [INFO] agent forwarding enabled
2022/01/31 16:51:33 ui: Connected to SSH!
2022/01/31 16:51:33 Running the provision hook
2022/01/31 16:51:33 ui: Stopping Compute instance
2022/01/31 16:51:39 ui: Creating Compute instance snapshot
2022/01/31 16:52:07 ui: Exporting Compute instance snapshot
2022/01/31 16:52:10 ui: Registering Compute instance template
2022/01/31 16:53:17 ui: Cleanup: destroying Compute instance
2022/01/31 16:53:23 ui: Cleanup: deleting SSH key
--- PASS: TestAccBuilder (158.51s)
=== RUN   TestSuiteExoscaleBuilder
=== RUN   TestSuiteExoscaleBuilder/TestArtifact_Destroy
=== RUN   TestSuiteExoscaleBuilder/TestBuilder_Prepare
=== RUN   TestSuiteExoscaleBuilder/TestBuilder_Prepare/fail_missing_required_config_args
=== RUN   TestSuiteExoscaleBuilder/TestBuilder_Prepare/ok
=== RUN   TestSuiteExoscaleBuilder/TestNewConfig
=== RUN   TestSuiteExoscaleBuilder/TestStepCreateInstance_Cleanup
=== RUN   TestSuiteExoscaleBuilder/TestStepCreateInstance_Run
=== RUN   TestSuiteExoscaleBuilder/TestStepCreateSSHKey_Cleanup
=== RUN   TestSuiteExoscaleBuilder/TestStepCreateSSHKey_Run
=== RUN   TestSuiteExoscaleBuilder/TestStepExportSnapshot_Run
=== RUN   TestSuiteExoscaleBuilder/TestStepRegisterTemplate_Run
=== RUN   TestSuiteExoscaleBuilder/TestStepSnapshotInstance_Run
=== RUN   TestSuiteExoscaleBuilder/TestStepStopInstance_Run
--- PASS: TestSuiteExoscaleBuilder (0.03s)
    --- PASS: TestSuiteExoscaleBuilder/TestArtifact_Destroy (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestBuilder_Prepare (0.00s)
        --- PASS: TestSuiteExoscaleBuilder/TestBuilder_Prepare/fail_missing_required_config_args (0.00s)
        --- PASS: TestSuiteExoscaleBuilder/TestBuilder_Prepare/ok (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestNewConfig (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepCreateInstance_Cleanup (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepCreateInstance_Run (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepCreateSSHKey_Cleanup (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepCreateSSHKey_Run (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepExportSnapshot_Run (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepRegisterTemplate_Run (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepSnapshotInstance_Run (0.00s)
    --- PASS: TestSuiteExoscaleBuilder/TestStepStopInstance_Run (0.00s)
PASS
ok      github.com/exoscale/packer-plugin-exoscale/builder/exoscale     158.592s
=== RUN   TestNewConfig
--- PASS: TestNewConfig (0.00s)
=== RUN   TestAccPostProcessor
2022/01/31 16:50:51 ui: Uploading template image
2022/01/31 16:50:52 ui: Registering Compute instance template
2022/01/31 16:50:56 ui: Deleting uploaded template image
--- PASS: TestAccPostProcessor (7.55s)
=== RUN   TestSuiteExoscaleImportPostProcessor
=== RUN   TestSuiteExoscaleImportPostProcessor/TestArtifact_Destroy
=== RUN   TestSuiteExoscaleImportPostProcessor/TestStepRegisterTemplate_Run
=== RUN   TestSuiteExoscaleImportPostProcessor/TestStepUploadImage_Cleanup
=== RUN   TestSuiteExoscaleImportPostProcessor/TestStepUploadImage_Run
--- PASS: TestSuiteExoscaleImportPostProcessor (0.02s)
    --- PASS: TestSuiteExoscaleImportPostProcessor/TestArtifact_Destroy (0.00s)
    --- PASS: TestSuiteExoscaleImportPostProcessor/TestStepRegisterTemplate_Run (0.00s)
    --- PASS: TestSuiteExoscaleImportPostProcessor/TestStepUploadImage_Cleanup (0.00s)
    --- PASS: TestSuiteExoscaleImportPostProcessor/TestStepUploadImage_Run (0.00s)
PASS
ok      github.com/exoscale/packer-plugin-exoscale/post-processor/exoscale-import       7.628s
```

### CODE REVIEW

File: builder/exoscale/config.go [hunk 1]
```diff
@@ -90,6 +91,10 @@ func NewConfig(raws ...interface{}) (*Config, error) {
 		return nil, errs
 	}
 
+	if config.APITimeout == 0 {
+		config.APITimeout = 600
```
Comment (arnaudgeiser) [inline builder/exoscale/config.go:95]: In my opinion, this value should match the one from the `post-processor`.
What takes (too) long is registering template and it's the associated comment.

```go
// Template registration can take a _long time_, raising
// the Exoscale API client timeout as a precaution.
egoscale.ClientOptWithTimeout(time.Hour),
```

Comment (arnaudgeiser) [reply builder/exoscale/config.go]: Thanks a lot!

**Review by arnaudgeiser — APPROVED** [review]

File: docs/builders/exoscale.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ other OS, we recommend using the [QEMU][packerqemu] plugin combined with the
 
 ### Optional
 
+- `api_timeout` (int) - The maximum API async operations waiting time in seconds.
+  Default to `3600`.
```
Comment (arnaudgeiser) [inline docs/builders/exoscale.md:33]:
```suggestion
  Defaults to `3600`.
```

File: docs/post-processors/exoscale-import.md [hunk 3]
```diff
@@ -27,6 +27,9 @@ built locally by the [`qemu`][packer-doc-builder-qemu] builder.
 
 ### Optional
 
+- `api_timeout` (int) - The maximum API async operations waiting time in seconds.
+  Default to `3600`.
```
Comment (arnaudgeiser) [inline docs/post-processors/exoscale-import.md:31]:
```suggestion
  Defaults to `3600`.
```

**Review by irregulator — APPROVED** [review]
lgtm, thanks!

**Pull request merged**